### ISSUE TITLE
Quantity defaults to latex output in ipython (Spyder), which is unreadable

### ISSUE THREAD

**Issue by MrBudgens** [astropy/astropy#11209]
**EDIT: Workaround found at** https://github.com/astropy/astropy/issues/11209#issuecomment-754329578

Starting Spyder, and typing anything along the following lines into the ipython console...
```python
>>>from astropy import units as u
>>>1 * u.K
```
...produces unreadable garbage because it defaults to Latex, which manifests as poorly-rendered white outlines on a black background. To add insult to injury, the rendering takes approximately 1 second.

The "workaround" is to write e.g. "str(1 * u.K)", but that's unacceptable because it becomes very tedious very quickly when examining numerous quantities.

After extensive searching I cannot find any way to control the default behaviour.

**Comment by mhvk** [comment]
Hmm, weird! Like many classes, `Quantity` defines a `_repr_latex_` which ipython can use, and this method has been the same nearly forever (it works fine on `jupyter-notebook`). I would think there would be some ipython option to just use the regular `__repr__`. Do other non-astropy classes that use `_repr_latex_` work fine?

**Comment by MrBudgens** [comment]
I'm not familiar with any other classes that use that function, so it's hard to know for sure, but so far the problem has been exclusive to astropy Quantitys. Also I can't see an easy way to choose which __repr__ function ipython uses by default.

I should mention that the issue has consistently occurred (for me) across platforms and versions over the last few years: I have just tested a Windows 10 machine, Ubuntu on the same machine via WSL, and a Scientific Linux machine. Both linux installations render black text on an almost-black background, which is not quite as fatal but still undesirable.

**Comment by ayshih** [comment]
Yeesh, [this Stack Overflow thread](https://stackoverflow.com/questions/48554207/use-plain-text-widget-for-spyders-ipython-console) indicates that there is no way to configure Spyder to not use `_repr_latex_()` (at least that's how I read the Spyder maintainer's answer), and a suggested workaround to an analogous issue with `pint` (by someone who is not the Spyder maintainer) is to *delete* `_repr_latex_()` when you know you're going to be working in Spyder.

**Comment by ayshih** [comment]
Yup, this works, but man that's unsatisfying:
```python
>>> from astropy import units as u
>>> delattr(u.Quantity, '_repr_latex_')
>>> 1 * u.K
```

**Comment by MrBudgens** [comment]
Confirmed, that works. Unsatisfying, but a much better solution than `str(...)`

Curiously `delattr(u.Quantity, '_repr_latex_')` complains with `AttributeError: _repr_latex_`, although that is less of a concern

**Comment by mhvk** [comment]
Yikes. Are the colours also not configurable? The latex string gets passed through something that turns it into an image, which may be configurable (`dvipng`??)

Just checked, and unlike what I seemed to remember, in astropy we indeed only use `_repr_latex_` for `Quantity` and its subclasses.

It still feels that this is an issue primarily about `Spider` not allowing control of typesetting; `Quantity` just provides an option to use a string that is latex-formatted, and really has no way to influence whether that string is used, what colours are taken, etc. Is there perhaps another `_repr_*` that would override it?

That said, if people can think of an easy way to control this (easy to maintain too...), then a PR would be welcome.

Short of that, it would be good to at least document the issue...

**Comment by mhvk** [comment]
Let me also ping @astrofrog - usually has good ideas on issues like these!

**Comment by pllim** [comment]
Re: Documenting -- Perhaps an entry in https://docs.astropy.org/en/latest/known_issues.html ?

**Comment by mhvk** [comment]
Please do raise an issue with Spyder! Ideally, they should really provide some control over whether `_repr_latex` gets used.

On the astropy side, I cannot think of great solutions, though perhaps one way would be to have a configuration item in `astropy.units.quantity` that, e.g., sets the format to be used in  `_repr_latex_`. A PR for that would be welcome. I think it would be relatively simple, just add a `repr_latex_format` to https://github.com/astropy/astropy/blob/0a29184ea1d5fdc2876e9dd3294c56e32287e0a5/astropy/units/quantity.py#L46-L55

and then use it in https://github.com/astropy/astropy/blob/0a29184ea1d5fdc2876e9dd3294c56e32287e0a5/astropy/units/quantity.py#L1238-L1248

(With some care with `subfmt`; maybe the configuration item should be a string with a comma separating a possible `subfmt`, i.e., 'latex,inline' by default)

**Comment by moux233** [comment]
Dear mhvk
Indeed I have raised an issue with Spyder.
https://github.com/spyder-ide/spyder/issues/14895

**Comment by JasonS09** [comment]
Hello. I'll work in this issue since it seems no one has taken it yet.

**Comment by mhvk** [comment]
@JasonS09 - before you do, have a look at the `spyder-ide` issue linked to above - my sense was that the problem was going to be solved there (and at least a work-around was given). As you'll have seen from the discussion above, I do not really think color selection of output text belongs in astropy! I certainly don't particularly like the suggested solution I gave; it just seemed there might not be an alternative...

**Comment by JasonS09** [comment]
@mhvk  So, do you think we should better wait for the `spyder-ide` issue resolution?

**Comment by mhvk** [comment]
@JasonS09 - yes, the problem does seem with them not dealing properly with a dark background, and not allowing one to avoid the use of `_repr_latex`.

**Comment by ccordoba12** [comment]
This should be fixed in Spyder 5.3.0, to be released at the end of March. Please test it when it's out and let us know if that's the case.
